src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/OEPublicDataRetrievalHelper.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+
+import java.io.IOException;
+
+@Component
+public class OEPublicDataRetrievalHelper {
+    private final ApiClientService apiClientService;
+
+    public OEPublicDataRetrievalHelper(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void getOverseasEntityPublicData(String companyReference, String passThroughTokenHeader) throws ServiceException {
+        getCompanyProfile(companyReference, passThroughTokenHeader);
+    }
+    private CompanyProfileApi getCompanyProfile(String companyReference, String passThroughTokenHeader) throws ServiceException {
+        try {
+            CompanyProfileApi companyProfileData = apiClientService
+                    .getOauthAuthenticatedClient(passThroughTokenHeader).company().get("/company/" + companyReference).execute().getData();
+            ApiLogger.info("Retrieving Company Profile Number: " + companyProfileData.getCompanyNumber());
```

Comment:
Please use log map so that company number is searchable. Need to get a key for company number from structured logging for that though. They are listed on confluence somewhere ...

Reply:
"Retrieving company profile for company number: " (you re not retrieving the company profile number)